File: 0160-0169/0168.rb
steps = [1, 1, 2, 4]
(4..30).to_a.each do |i|
  steps[i] = steps[i-1] + steps[i-2] + steps[i-3]
end

while gets
  n = $_.to_i
  break if n == 0
  puts (steps[n]/3650.0).ceil
end

File: python/0168.py
import math

steps=[1, 1, 2, 4]+ [0]*27
for i in range(4, 31):
  steps[i] = steps[i-1] + steps[i-2] + steps[i-3]

while True:
  n = input()
  if n == 0: break
  print int(math.ceil(steps[n]/3650.0))